src/utils.ts

```diff
@@ -20,4 +21,31 @@ export const camelCaseKeys = (rawFlags: LDFlagSet) => {
   return flags;
 };
 
-export default { camelCaseKeys };
+/**
+ * Gets the flags to pass to the provider from the changeset.
+ *
+ * @param changes the `LDFlagChangeset` from the ldClient onchange handler.
+ * @param targetFlags if targetFlags are specified, changes to other flags are ignored and not returned in the
+ * flattened `LDFlagSet`
+ * @param reactOptions reactOptions.useCamelCaseFlagKeys determines whether to change the flag keys to camelCase
+ * @return an `LDFlagSet` with the current flag values from the LDFlagChangeset filtered by `targetFlags`. The returned
+ * object may be empty `{}` if none of the targetFlags were changed.
+ */
+export const getFlattenedFlagsFromChangeset = (
+  changes: LDFlagChangeset,
+  targetFlags: LDFlagSet | undefined,
+  reactOptions: LDReactOptions,
+): LDFlagSet => {
+  const flattened: LDFlagSet = {};
+  for (const key in changes) {
+    if (targetFlags === undefined || targetFlags[key] !== undefined) {
```

Comment:
You can simplify these comparisons with `undefined` to the following equivalent:

```tsx
if (!targetFlags || targetFlags[key]) {
```

Reply:
I think that would fail to add changed flags to the `flattened` flags map for any targetFlags that were defined as boolean flags with default value of false.

```
const targetFlags = {
     testFlag: false
}
```

Now anytime a change happens to `testFlag` that change won't be pushed to the provider because `targetFlags['testFlag']` will be false.

Reply:
You're right; I found the same thing when drafting this change on our private SDK mirror. The following simplification is more appropriate:

```
if (!targetFlags || targetFlags[key] !== undefined) {
```

I have this change PR'd on our private mirror and will proceed with the release once this is merged.